### DISCUSSION TITLE
Which method can I use to...

### DISCUSSION THREAD

**Discussion by ricardodacosta** [amzn/ads-advanced-tools-docs#125] [Q&A]
I would like to be able to get bid suggestion ranges for Sponsored Product campaigns targeting both ASINS and KEYWORDS and use such data to then send another request to update bids on my campaigns.

I have read through the documentation and is not clear to me which method to use to update bids on previously created campaigns nor am able to find a method to get bid suggestion in decimal format. So far I have seen only integers in the hundreds. (I submitted another question about that in particular)

If someone can point me to the correct methods I should choose, I will appreciate it

Thanks in advance

~ Ricardo

**Comment by elizafarley** [comment]
Hi Ricardo, thanks for your questions!

First, to update bids for existing campaigns, you can use [PUT /sp/targets](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#tag/TargetingClauses/operation/UpdateSponsoredProductsTargetingClauses) or [PUT /sp/keywords](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#tag/Keywords/operation/UpdateSponsoredProductsKeywords) to update the targets for individual product or keyword targets. You can also update the default bid for an ad group using [PUT /sp/adGroups](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#tag/AdGroups/operation/UpdateSponsoredProductsAdGroups). 

In the other thread, I said that what you are experiencing with that particular endpoint is likely a bug, but we do have a couple of other bid recommendations endpoints you can try. 

[Theme-based bid recommendations](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#tag/ThemeBasedBidRecommendation) - Provides bid suggestions for keywords only. Note that it also has some geographic limitations. 
[General bid recommendations](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#tag/ThemeBasedBidRecommendation) - Provides bid recommendations for both keywords and targets for existing ad groups.

Let me know if that answers your question! Happy to provide code samples if needed.
